www/webapp/templates/signup.html

```diff
@@ -0,0 +1,249 @@
+          signUpForm[i].classList.remove("d-none");
+          requiredSignUpInputs();
+        }
+        checkedEmail = true;
+        btnSubmit.innerText = "Sign Up";
+        mainTitle.innerText = "Please fill the sign up form";
+      }
+
+      if (response.ok) {
+        for (let i = 0; i < signInForm.length; i++) {
+          signInForm[i].classList.remove("d-none");
+          requiredSignInInput();
+        }
+        checkedEmail = true;
+        existUser = true;
+        btnSubmit.innerText = "Sign In";
+        mainTitle.innerText = "Please enter your password";
+      }
+    }
+    async function loginForm() {
+      const email = document.getElementById("email").value;
+      const password = document.getElementById("password").value;
+      const errorMessage = document.getElementById("errorMessage");
+
+      errorMessage.classList.add("d-none");
+
+      const formData = {
+        email: email,
+        password: password,
+      };
+
+      const response = await fetch("/api/auth/sign_in", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(formData),
+      });
+
+      if (!response.ok) {
+        errorMessage.classList.remove("d-none");
+        return;
+      }
+
+      const data = await response.json();
+      const token = data.token;
+      if (token) {
+        sessionStorage.setItem("token", token);
+        window.location.href = "/";
+      }
+    }
+
+    async function registerForm() {
+      const given_name = document.getElementById("given_name").value;
+      const surname = document.getElementById("surname").value;
+      const email = document.getElementById("email").value;
+      const password = document.getElementById("password").value;
+      const errorRegisterMessage = document.getElementById(
+        "errorRegisterMessage"
+      );
+
+      errorRegisterMessage.classList.add("d-none");
+
+      const formData = {
+        email: email,
+        password: password,
+        given_name: given_name,
+        surname: surname,
+      };
+
+      const response = await fetch("/api/auth/sign_up", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(formData),
+      });
+
+      if (!response.ok) {
+        errorRegisterMessage.classList.remove("d-none");
+        return;
+      }
+      if (response.ok) {
+        loginForm();
+      }
+    }
+    document.querySelector("form").addEventListener("submit", function (event) {
+      event.preventDefault(); // Prevent the default form submission
+      switch (true) {
+        case !checkedEmail:
+          checkMail();
+          break;
+        case existUser && checkedEmail:
+          loginForm();
+          break;
+        case !existUser && checkedEmail:
+          registerForm();
+          break;
+        default:
+          break;
+      }
+    });
```

Comment:
I think this can be manage by a ENUM and a tag on the button instead of multiple flags.
```suggestion
function changeSubmitButtonAction(action) {
   const actionButton = document.getActionButton();
   
   actionButton.attr('button-action', action);
}

const SubmitActions = {
    CHECK_EMAIL: 0,
    LOGIN: 1,
    REGISTER: 2
}

submitButton.addEventListener("submit", function (event) {
   const actions = {
     SubmitActions.CHECK_EMAIL: checkEmail,
     SubmitActions.LOGIN: loginForm,
     SubmitActions.REGISTER: registerForm
   }
   const currentAction = submitButton.getAttribute("button-action")
   const actionFunc = actions[`${currentAction}`] || () => { diplay error action not found}
   
   return actuonFunc()
}
```